samples: npm2100_fuel_gauge: Fuel gauge sample added (#20693)
nrfconnect/sdk-nrf: samples/pmic/native/npm2100_fuel_gauge/README.rst

.. _npm2100_fuel_gauge:

nPM2100: Fuel gauge
###################

.. contents::
   :local:
   :depth: 2

The Fuel gauge sample demonstrates how to calculate the state of charge of a supported primary cell battery using the `nPM2100 <nPM2100 product website_>`_ and the :ref:`nrfxlib:nrf_fuel_gauge`.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

The sample also requires an nPM2100 evaluation kit that you need to connect to the development kit as described in `Wiring`_.

Overview
********

This sample allows the calculation of state of charge from a battery connected to the nPM2100 PMIC.

Battery models for Alkaline AA, AAA, LR44, and Lithium-manganese dioxide coin cell CR2032 batteries are included.
You can change the active battery model using a shell command, as illustrated in `Testing`_.

> The battery model can also be changed at compile time using the :kconfig:option:`CONFIG_BATTERY_MODEL` option.

Reviewer: ```suggestion
You can also change the battery model at compile time using the :kconfig:option:`CONFIG_BATTERY_MODEL` Kconfig option.
```

.. _npm2100_fuel_gauge:

nPM2100: Fuel gauge
###################

.. contents::
   :local:
   :depth: 2

> The fuel gauge sample demonstrates how to calculate the state of charge of a supported primary cell battery using `nPM2100 <nPM2100 product website_>`_ and the :ref:`nrfxlib:nrf_fuel_gauge`.

Reviewer: ```suggestion
The fuel gauge sample demonstrates how to calculate the state of charge of a supported primary cell battery using the `nPM2100 <nPM2100 product website_>`_ and the :ref:`nrfxlib:nrf_fuel_gauge`.
```

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

The sample also requires an nPM2100 evaluation kit that you need to connect to the development kit as described in `Wiring`_.

Overview
********

This sample allows the calculation of state of charge from a battery connected to the nPM2100 PMIC.

> Battery models for Alkaline AA, AA, LR44, and Lithium-manganese dioxide coin cell CR2032 batteries are included. The active battery model can be changed using a shell command, as illustrated in `Testing`_.

Reviewer: ```suggestion
Battery models for Alkaline AA, AA, LR44, and Lithium-manganese dioxide coin cell CR2032 batteries are included.
You can change the active battery model using a shell command, as illustrated in `Testing`_.
```

Wiring
******

With this configuration, the nPM2100 EK is wired to supply power to the DK.
This ensures that the TWI communication is at compatible voltage levels, and represents a realistic use case for the nPM2100 PMIC.

.. note::

>    In order prevent leakage currents and to continue being able to program the DK, do not remove the DK USB connection.

Reviewer: ```suggestion
   To prevent leakage currents and program the DK, do not remove the USB connection.
```

.. note::

>    Unplug the battery from nPM2100 EK, and set the DK power switch to "OFF" while applying the wiring described below.
>    In case of issues communicating with or programming the DK after applying the wiring, try to power cycle the DK and EK.

Reviewer: ```suggestion
   Unplug the battery from the nPM2100 EK and set the DK power switch to "OFF" while applying the wiring described below.
   If you have issues communicating with the DK or programming it after applying the wiring, try to power cycle the DK and EK.
```

Reviewer: How about combining the two consecutive notes to one?

Author: I'm fine with that, if you think it is better from a style perspective. The reason for having them separate is that they cover two slightly different things. But I can have two small paragraphs in the same note

   * On the **BOOTMON** pin header, select **OFF** with a jumper.
   * On the **VSET** pin header, select **3.0V** with a jumper.
   * On the **VBAT SEL** switch, select **VBAT** position.
   * Connect a battery board to the **BATTERY INPUT** connector.

Building and running
********************

.. |sample path| replace:: :file:`samples/pmic/native/npm2100_fuel_gauge`

.. include:: /includes/build_and_run.txt

Testing
*******

|test_sample|

#. |connect_kit|
#. |connect_terminal|

If the initialization was successful, the terminal displays the following message with status information:

.. code-block:: console

   PMIC device ok
   nRF Fuel Gauge version: 1.0.0
   Fuel gauge initialised for Alkaline AA battery.
   V: 1.188, T: 20.62, SoC: 25.00

.. _table::
   :widths: auto

   ======  ===============  ==================================================
   Symbol  Description      Units
   ======  ===============  ==================================================
   V       Battery voltage  Volts
   T       Temperature      Degrees C
   SoC     State of Charge  Percent
   ======  ===============  ==================================================

> The active battery type can be determined by using the following shell command:

Reviewer: ```suggestion
You can determine the active battery type using the following shell command:
```

.. code-block:: console

   $ battery_model
   Battery model: Alkaline AA

> The same shell command can be used to change the active battery type:

Reviewer: ```suggestion
Use the same shell command to change the active battery type:
```

.. _npm2100_fuel_gauge:

nPM2100: Fuel gauge
###################

.. contents::
   :local:
   :depth: 2

> The fuel gauge sample demonstrates how to calculate the state of charge of a supported primary cell battery using the `nPM2100 <nPM2100 product website_>`_ and the :ref:`nrfxlib:nrf_fuel_gauge`.

Reviewer: ```suggestion
The Fuel gauge sample demonstrates how to calculate the state of charge of a supported primary cell battery using the `nPM2100 <nPM2100 product website_>`_ and the :ref:`nrfxlib:nrf_fuel_gauge`.
```

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

The sample also requires an nPM2100 evaluation kit that you need to connect to the development kit as described in `Wiring`_.

Overview
********

This sample allows the calculation of state of charge from a battery connected to the nPM2100 PMIC.

Battery models for Alkaline AA, AAA, LR44, and Lithium-manganese dioxide coin cell CR2032 batteries are included.
You can change the active battery model using a shell command, as illustrated in `Testing`_.

Wiring
******

With this configuration, the nPM2100 EK is wired to supply power to the DK.
This ensures that the TWI communication is at compatible voltage levels, and represents a realistic use case for the nPM2100 PMIC.

.. note::

   To prevent leakage currents and program the DK, do not remove the USB connection.

.. note::

   Unplug the battery from the nPM2100 EK and set the DK power switch to "OFF" while applying the wiring described below.
   If you have issues communicating with the DK or programming it after applying the wiring, try to power cycle the DK and EK.

> The following preparations should be made to the DK in order to power the nRF5x SoC from nPM2100 EK:

Reviewer: ```suggestion
Prepare the DK to power the nRF5x SoC from the nPM2100 EK as follows:
```

#. Prepare the DK for being powered by the nPM2100 EK:

>    **nRF52840 DK:**
> 
>    * Set switch **SW9** ("nRF power source") to position "VDD".
>    * Set switch **SW10** ("VEXT -> VnRF") to position "ON".
> 
>    **nRF5340 DK:**
> 
>    * Set switch **SW9** ("nRF power source") to position "VDD".
>    * Set switch **SW10** ("VEXT -> VnRF") to position "ON".
> 
>    **nRF54L15 DK:**
> 
>    * Remove jumper from **P6** ("VDDM CURRENT MEASURE").

Reviewer: ```suggestion
   .. tabs::
   
      .. group-tab:: nRF52840 DK
   
         * Set switch **SW9** ("nRF power source") to position "VDD".
         * Set switch **SW10** ("VEXT -> VnRF") to position "ON".

      .. group-tab:: nRF5340 DK

         * Set switch **SW9** ("nRF power source") to position "VDD".
         * Set switch **SW10** ("VEXT -> VnRF") to position "ON".

      .. group-tab:: nRF54L15 DK

         * Remove jumper from **P6** ("VDDM CURRENT MEASURE").
```

Reviewer: I believe tabs would be nice here.

   * On the **BOOTMON** pin header, select **OFF** with a jumper.
   * On the **VSET** pin header, select **3.0V** with a jumper.
   * On the **VBAT SEL** switch, select **VBAT** position.
   * Connect a battery board to the **BATTERY INPUT** connector.

Building and running
********************

.. |sample path| replace:: :file:`samples/pmic/native/npm2100_fuel_gauge`

.. include:: /includes/build_and_run.txt

Testing
*******

|test_sample|

#. |connect_kit|
#. |connect_terminal|

If the initialization was successful, the terminal displays the following message with status information:

.. code-block:: console

   PMIC device ok
   nRF Fuel Gauge version: 1.0.0
   Fuel gauge initialised for Alkaline AA battery.
   V: 1.188, T: 20.62, SoC: 25.00

.. _table::
   :widths: auto

   ======  ===============  ==================================================
   Symbol  Description      Units
   ======  ===============  ==================================================
   V       Battery voltage  Volts
   T       Temperature      Degrees C
   SoC     State of Charge  Percent
   ======  ===============  ==================================================

> You can determine the active battery type using the following shell command:

Reviewer: ```suggestion
Determine the active battery type using the following shell command:
```
